.circleci/config.yml

```diff
@@ -0,0 +1,105 @@
+
+version: 2.1
+orbs:
+  commitlint: conventional-changelog/commitlint@1.0.0
+
+executors:
+  default-executor:
+    docker:
+      - image: "cimg/node:lts"
+    working_directory: ~/project
+    resource_class: medium
+
+commands:
+  restore_test_cache:
+    steps:
+      - restore_cache:
+          keys:
+            - v1-deps-{{ .Environment.CIRCLE_JOB }}-{{ .Branch }}-{{ .Revision }}
+            - v1-deps-{{ .Environment.CIRCLE_JOB }}-{{ .Branch }}
+            - v1-deps-
+  save_test_cache:
+    steps:
+      - save_cache:
+          key: v1-deps-{{ .Environment.CIRCLE_JOB }}-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - node_modules
+            - ~/.npm
+  setup:
+    steps:
+      - run:
+          name: Setup
+          command: |
+            npm prune
+  test:
+    steps:
+      - run:
+          name: Test
+          command: |
+            npm run test
+  deploy:
+    steps:
+      - run:
+          name: Deploy
+          command: |
+            npm run semantic-release
+            
+jobs:
+  build-and-test:
+    executor: default-executor
+    steps:
+      - checkout
+      - restore_test_cache
+      - setup
+      - save_test_cache
+      - test
+      - persist_to_workspace:
+          root: ~/project
+          paths: .
+  deploy:
+    executor: default-executor
+    steps:
+      - attach_workspace:
+          at: ~/project
+      - deploy
+
+workflows:
+  commitlint:
+    when:
+      not:
+        or:
+          - equal: [ master, <<pipeline.git.branch>> ]
+          - equal: [ develop, <<pipeline.git.branch>> ]
+          - matches: { pattern: "^hotfix.*", value: <<pipeline.git.branch>> }
+          - matches: { pattern: "^release.*", value: <<pipeline.git.branch>> }
```

Comment:
I hadn't thought to filter this, but that's a good idea. I should add this to other repos!